### PR TITLE
Add LLaMA C++ Model

### PR DESCRIPTION
**Pull request by D-X-Y**
- Add a LLaMA C++ model at `langfun/core/llms/llama_cpp.py`
- Fix various minor pytype or annotation issues.

### CODE REVIEW

File: langfun/core/llms/llama_cpp.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  ] = None
+
+  name: Annotated[
+      str,
+      'The abbreviation for the LLaMA CPP-based model name.',
+  ] = ''
+
+  def _on_bound(self):
+    super()._on_bound()
+    if not self.url:
```
Comment (daiyip) [inline langfun/core/llms/llama_cpp.py:42]: why not set url: str in this case?

**Review by daiyip — DISMISSED** [review]
This is awesome. Thanks for adding the Llama C++ model, Xuanyi.

File: .flake8
Comment (daiyip) [inline .flake8:1]: copybara complained about missing license header in this file. I think we will have to remove it by now before I registered it with copybara.

File: langfun/core/llms/llama_cpp.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+@lf.use_init_args(['url'])
+class LlamaCppRemote(lf.LanguageModel):
+  """The remote LLaMA C++ model.
+
+  The Remote LLaMA C++ models can be launched via
+  https://github.com/ggerganov/llama.cpp/tree/master/examples/server
+  """
+
+  url: Annotated[
+      str | None,
```
Comment (daiyip) [inline langfun/core/llms/llama_cpp.py:31]: just `str`, then you don't need to check in `_on_bound`.

File: .flake8
Comment (D-X-Y) [reply .flake8:1]: Sure, deleted for now~

File: langfun/core/llms/llama_cpp.py (hunk 2 above)
Comment (D-X-Y) [reply langfun/core/llms/llama_cpp.py]: SG, done~

**Review by daiyip — APPROVED** [review]

**Pull request merged**